File: packages/elixir-sdk/test/adapters/context_test.exs
defmodule TreeDxSdk.ContextAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Context.build(client, "repo/a", %{})
    TreeDxSdk.Context.parse(client, "repo/a", %{})

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :post and &1.path == "/api/v1/repos/repo%2Fa/context/parse-ctx")
           )
  end
end

File: packages/ts-sdk/test/adapters/context.test.ts
import { describe, expect, it } from 'vitest';
import { ContextAdapter } from '../../src/treedx/index.js';
import { MockTransport } from './mock.js';

describe('ContextAdapter', () => {
  it('constructs context endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new ContextAdapter({ transport });
    await adapter.build('repo', {});
    await adapter.parse('repo', {});
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'POST /api/v1/repos/repo/context/build',
      'POST /api/v1/repos/repo/context/parse-ctx'
    ]);
  });
});
